File: android/src/main/kotlin/eu/kevin/kevin_flutter/entity/PaymentSessionConfigurationEntity.kt
package eu.kevin.kevin_flutter.entity

import kotlinx.serialization.Serializable

@Serializable
internal data class PaymentSessionConfigurationEntity(
    val paymentId: String,
    val paymentType: String,
    val preselectedCountry: String?,
    val disableCountrySelection: Boolean,
    val countryFilter: List<String>,
    val bankFilter: List<String>,
    val preselectedBank: String?,
    val skipBankSelection: Boolean,
    val skipAuthentication: Boolean,
)

File: packages/kevin_flutter_in_app_payments/ios/Classes/Entity/PaymentSessionConfigurationEntity.swift
internal struct PaymentSessionConfigurationEntity : Codable {
    let paymentId: String
    let paymentType: String
    let preselectedCountry: String?
    let disableCountrySelection: Bool
    let countryFilter: [String]
    let bankFilter: [String]
    let preselectedBank: String?
    let skipBankSelection: Bool
    let skipAuthentication: Bool
    
    init(
        paymentId: String,
        paymentType: String,
        preselectedCountry: String?,
        disableCountrySelection: Bool,
        countryFilter: [String],
        bankFilter: [String],
        preselectedBank: String?,
        skipBankSelection: Bool,
        skipAuthentication: Bool) {
            self.paymentId = paymentId
            self.paymentType = paymentType
            self.preselectedCountry = preselectedCountry
            self.disableCountrySelection = disableCountrySelection
            self.countryFilter = countryFilter
            self.bankFilter = bankFilter
            self.preselectedBank = preselectedBank
            self.skipBankSelection = skipBankSelection
            self.skipAuthentication = skipAuthentication
        }
}
